### PR TITLE
Support Cake 2.0.0

### LINKED ISSUE #108: Support Cake 2.0.0
Please provide a version supporting Cake 2.0

### PR DESCRIPTION
**Pull request by twenzel**
- Update references to Cake 2.0.0
- Remove support for .NET Framework 4.6
- Support .NET 6
- Use Cake 2.0

Fixes #108
Fixes #107 
Fixes #106

### CODE REVIEW

File: src/Cake.Sonar.Test/Cake.Sonar.Test.csproj [hunk 1]
```diff
@@ -1,19 +1,19 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (augustoproiete) [inline src/Cake.Sonar.Test/Cake.Sonar.Test.csproj:4]:
```suggestion
    <TargetFrameworks>net6.0;net5.0;netcoreapp3.1</TargetFrameworks>
```

I'd suggest running tests on all frameworks that are being targeted by the Cake.Sonar addin

**Review by augustoproiete — APPROVED** [review]
LGTM

File: src/Cake.Sonar.Test/Cake.Sonar.Test.csproj (hunk 1 above)
Comment (twenzel) [reply src/Cake.Sonar.Test/Cake.Sonar.Test.csproj]: Done.
I also upgraded to Cake 2.0 and fixed the NuGet package creation.

**Comment by augustoproiete** [comment]
Thanks @twenzel LGTM. Will leave it for @tomstaijen to review and next steps

**Comment by HofmeisterAn** [comment]
We need an updated package too. It looks like a few people need it as well, but changes are not accepted. I do not know if it is the right approach it, but if you create a pull request on my [fork](https://github.com/HofmeisterAn/Cake.Sonar/pull/1), I can take care about the pipeline and release. We only need a different name...

**Comment by mcliment** [comment]
I was wondering that too but I think I'd be better to take ownership of this repo instead of doing forks. I'll try to contact the administrators of cake-contrib to see if I can do that.

**Comment by tomstaijen** [comment]
I'm happy to transfer my role ownership to a new maintainer person/team. I don't have bandwidth to support this as I currently not use .NET nor sonar.

**Comment by twenzel** [comment]
@tomstaijen great to hear. Who is the new maintainer person/team? We need to fix the CI build.

**Comment by twenzel** [comment]
Either we switch back to `Visual Studio 2022` image or find a solution for GitVersion...

**Comment by HofmeisterAn** [comment]
> I'm happy to transfer my role ownership to a new maintainer person/team. I don't have bandwidth to support this as I currently not use .NET nor sonar.

@tomstaijen as @twenzel already asked, who is the new maintainer person / team? It would be nice if we can keep the package up to date. I'm happy to help and join the team, if there is already one.

**Comment by tomstaijen** [comment]
Mark Climent has been added as maintainer. I'll tell them to add you too!

https://github.com/cake-build/cake/discussions/3738

**Pull request merged**